unit/util/simplify_expr.cpp

```diff
@@ -70,13 +73,35 @@ TEST_CASE("Simplify Java boolean -> int -> boolean casts")
       typecast_exprt(
         symbol_exprt(
           "foo",
-          java_boolean_type()),
+          get_type_func()),
         java_int_type()),
-      java_boolean_type()),
+      get_type_func()),
     namespacet(symbol_tablet()));
 
   REQUIRE(simplified.id()==ID_symbol);
-  REQUIRE(simplified.type()==java_boolean_type());
+  REQUIRE(simplified.type()==get_type_func());
   const auto &symbol=to_symbol_expr(simplified);
   REQUIRE(symbol.get_identifier()=="foo");
 }
+
+} // namespace
+
+TEST_CASE("Simplify Java boolean -> int -> boolean casts")
```

Comment:
You should add tests for the cases that shouldn't be simplified ideally (e.g. a typecast from an int to a short)